apps/devhub/views.py

```diff
@@ -1386,20 +1385,24 @@ def _step_url(step):
 
 @login_required
 @submit_step(1)
-def submit(request, step):
+def submit(request, step, next_step=_step_url(2)):
     if request.method == 'POST':
         request.user.update(read_dev_agreement=now())
-        response = redirect(_step_url(2))
-        response.set_cookie(DEV_AGREEMENT_COOKIE)
+        response = redirect(next_step)
         return response
 
-    return render(request, 'devhub/addons/submit/start.html', {'step': step})
+    if request.user.read_dev_agreement is None:
+        return render(request, 'devhub/addons/submit/start.html',
+                      {'step': step})
+    else:
+        response = redirect(next_step)
+        return response
 
 
 @login_required
 @submit_step(2)
 def submit_addon(request, step):
-    if DEV_AGREEMENT_COOKIE not in request.COOKIES:
```

Comment:
thanks, I think this makes everything more straightforward. Please also rename the old tests that have '..with_cookie' in the names to reflect what they are actually testing now.